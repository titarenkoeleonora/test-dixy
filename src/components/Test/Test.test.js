import Test from './Test';
import renderer from 'react-test-renderer';

const testQuestions = [
  {
    id: '00',
    title: 'СПИСОК ПОКУПОК – ЭТО…',
    answers: [{
      answer: 'Список чего? У меня все есть.',
      variant: 'a',
    }, 
    {
      answer: 'Кандалы, которые сковывают порывы моей души. Надо покупать сердцем, а не списком.',
      variant: 'b',
    }, 
    {
      answer: 'Огонёк маяка, который ведет меня сквозь гастрономический шторм.',
      variant: 'c',
    }],
  }, 
  {
    id: '00',
    title: 'ПОЛЬЗУЕШЬСЯ ЛИ ТЫ СКИДОЧНЫМИ КУПОНАМИ?',
    answers: [
      {
      answer: 'Конечно! А ещё промокодами и дисконтными картами',
      variant: 'a',
    }, 
    {
      answer: 'Зачем платить больше, когда можно то же самое купить со скидкой.',
      variant: 'b',
    }, 
    {
      answer: 'Слишком много суеты: нужно их где-то найти, потом не забыть обменять…',
      variant: 'c',
    }],
  }, 
];

const testResults = [
  {
    id: 'a',
    title: 'НЕДОВЕР НЕДОВЕРЫЧ',
    text: 'Ты знаешь марку самой натуральной колбасы, а также где максимально выгодно ее купить. Твой список покупок всегда наготове, и ты умело им распоряжаешься, не теряя голову при виде акционных товаров. Мудрость и рациональность в продуктовом шопинге – твоя отличительная черта, это очень ценные качества, береги их.',
  }, {
    id: 'b',
    title: 'СКИДУЛЬКИН',
    text: 'При виде слова «скидка» твои ладони начинают потеть, а перед глазами появляется легкая дымка. Ты носишься по рядам магазина, сбрасывая в корзинку всё, что попадается под руку. Ты часто набираешь больше, чем можешь съесть, стараясь оправдать себя тем, что это можно отдать соседу, а это точно пригодится. В этом нет ничего ужасного, ты любишь продуктовый шопинг, но попробуй хоть иногда воспользоваться списком.',
  }, {
    id: 'c',
    title: 'УМЕЛОВ',
    text: 'Ты очень бережлив и осторожен. Акции для тебя пустой звук, ты относишься к ним с недоверием. Но попробуй открыться продуктовому миру – он не такой хитрый, как тебе кажется. Постарайся иногда расслабиться и получить удовольствие от покупок, как это делают многие клиенты «Дикси».',
  }, 
];

it(`Correctly Test render`, () => {
  const tree = renderer
    .create(
      <Test
        questions={testQuestions}
        testResults={testResults}
      />
    );

  expect(tree).toMatchSnapshot();
});