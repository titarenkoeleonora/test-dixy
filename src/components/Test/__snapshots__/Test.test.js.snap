// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly Test render 1`] = `
<section
  className="test"
>
  <div
    className="container"
  >
    <div
      className="test__wrapper"
    >
      <div
        className="test__text-wrapper"
      >
        <h2
          className="test__title"
        >
          Пройди тест
        </h2>
        <p
          className="test__text"
        >
          Значимость этих проблем настолько очевидна, что постоянный количественный рост и сфера нашей активности позволяет оценить значение новых предложений.
        </p>
      </div>
      <form
        className="form-test test__form"
        method="post"
        onSubmit={[Function]}
      >
        <div>
          <p
            className="form-test__question"
          >
            СПИСОК ПОКУПОК – ЭТО…
          </p>
          <ul
            className="form-test__answers-list"
          >
            <li
              className="form-test__answers-item"
            >
              <input
                className="form-test__answer-input checkbox visually-hidden"
                id="010"
                name="question-list"
                required={true}
                type="radio"
                value="a"
              />
              <label
                className="form-test__answer-text"
                htmlFor="010"
              >
                Список чего? У меня все есть.
              </label>
            </li>
            <li
              className="form-test__answers-item"
            >
              <input
                className="form-test__answer-input checkbox visually-hidden"
                id="011"
                name="question-list"
                required={true}
                type="radio"
                value="b"
              />
              <label
                className="form-test__answer-text"
                htmlFor="011"
              >
                Кандалы, которые сковывают порывы моей души. Надо покупать сердцем, а не списком.
              </label>
            </li>
            <li
              className="form-test__answers-item"
            >
              <input
                className="form-test__answer-input checkbox visually-hidden"
                id="012"
                name="question-list"
                required={true}
                type="radio"
                value="c"
              />
              <label
                className="form-test__answer-text"
                htmlFor="012"
              >
                Огонёк маяка, который ведет меня сквозь гастрономический шторм.
              </label>
            </li>
          </ul>
          <button
            aria-label="Далее"
            className="form-test__button"
            type="submit"
          >
            Далее
          </button>
        </div>
      </form>
    </div>
  </div>
</section>
`;
